src/super_gradients/training/models/pose_estimation_models/yolo_nas_pose/yolo_nas_pose_ndfl_heads.py

```diff
@@ -0,0 +1,220 @@
+from typing import Tuple, Union, List, Callable, Optional
+
+import torch
+from omegaconf import DictConfig
+from torch import nn, Tensor
+
+import super_gradients.common.factories.detection_modules_factory as det_factory
+from super_gradients.common.registry import register_detection_module
+from super_gradients.module_interfaces import SupportsReplaceNumClasses
+from super_gradients.modules.base_modules import BaseDetectionModule
+from super_gradients.training.models.detection_models.pp_yolo_e.pp_yolo_head import generate_anchors_for_grid_cell
+from super_gradients.training.utils import HpmStruct, torch_version_is_greater_or_equal
+from super_gradients.training.utils.bbox_utils import batch_distance2bbox
+from super_gradients.training.utils.utils import infer_model_dtype, infer_model_device
+
+
+@register_detection_module()
+class YoloNASPoseNDFLHeads(BaseDetectionModule, SupportsReplaceNumClasses):
+    def __init__(
+        self,
+        num_classes: int,
+        in_channels: Tuple[int, int, int],
+        heads_list: Union[str, HpmStruct, DictConfig],
```

Comment:
I generally agree that we should allow passing already instantiated heads. 
However in current design of the entire model instantiation I don't see a good way to support this 🤷 
Perhaps, a right way to approach this would be an introduction of classmethods `from_config` to discriminate the intention:

```python
class YoloNASPoseNDFLHeads(BaseDetectionModule, SupportsReplaceNumClasses):
  def __init__(
    self,
    heads_list: List[YoloNASPoseHead]
  ):
    self.heads_list = heads_list
  
  @classmethod
  def from_config(cls, config):
    heads_list = INSTANTIATE HEADS FROM CONFIG
    return cls(heads_list)
```

But again, this is pretty massive refactor and I'm not sure how to approach this now